presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -970,52 +970,7 @@ else if (criteria instanceof JoinOn) {
 
                 Analyzer.verifyNoAggregatesOrWindowFunctions(metadata.getFunctionRegistry(), expression, "JOIN clause");
 
-                Set<Expression> postJoinConjuncts = new HashSet<>();
-
-                for (Expression conjunct : ExpressionUtils.extractConjuncts(expression)) {
-                    conjunct = ExpressionUtils.normalize(conjunct);
```

Comment:
The original code (commit id: b34d126482a9916c5873a6b945538e055ec4dbd7) is quite old (2015). The  purpose was to verify the join criteria (no non-equi joins). This is no longer required since join can have arbitrary filter expression. 

I think the normalization was to have support for conditions like `not (a != b)`. The normalization step is repeated in `RelationPlanner#visitJoin`